ee/tabby-ui/components/prompt-form.tsx

```diff
@@ -26,15 +35,56 @@ export function PromptForm({
   isLoading
 }: PromptProps) {
   const { formRef, onKeyDown } = useEnterSubmit()
-  const inputRef = React.useRef<HTMLTextAreaElement>(null)
-  const router = useRouter()
+  const [queryCompletionUrl, setQueryCompletionUrl] = React.useState<
+    string | null
+  >(null)
+  const latestFetchKey = React.useRef('')
 
-  React.useEffect(() => {
-    if (inputRef.current) {
-      inputRef.current.focus()
+  useSWR<SearchReponse>(queryCompletionUrl, fetcher, {
+    revalidateOnFocus: false,
+    dedupingInterval: 500,
+    onSuccess: (data, key) => {
+      if (key !== latestFetchKey.current) return
+
+      setOptions(data?.hits ?? [])
     }
+  })
+
+  const [options, setOptions] = React.useState<SearchReponse['hits']>([])
+  const onSearch = React.useMemo(() => {
+    return debounce((e: React.ChangeEvent<HTMLTextAreaElement>) => {
+      const value = e.target?.value ?? ''
+      const end = e.target?.selectionEnd ?? 0
+      const queryname = getSearchCompletionQueryName(value, end)
+      if (queryname) {
+        let url = `/v1beta/search?q=name:${queryname}`
+        latestFetchKey.current = url
+        setQueryCompletionUrl(`/v1beta/search?q=name:${queryname}`)
+      } else {
+        setOptions([])
+      }
+    }, 200)
   }, [])
 
+  const onSelectCompletion = (
+    inputRef: React.MutableRefObject<
+      HTMLTextAreaElement | HTMLInputElement | null
+    >,
+    item: ISearchHit
+  ) => {
+    const replaceString = '`@' + item?.doc?.name + '` '
```

Comment:
no need to replace @get_name, just keep it as is.

Reply:
keep it as is and wrap with `` ?

Reply:
keep it as is and wrap with `` ?

Reply:
i'll so it's not necessary to wrap with ``. But certain highlighting on `@get_function` will be somewhat helpful (not required in this PR)

Reply:
Done. And a space is inserted after the input to break the query string.